app/code/Magento/Catalog/Block/Product/View/Details.php

```diff
@@ -30,18 +30,29 @@ public function getGroupSortedChildNames(string $groupName, string $callback): a
     {
         $groupChildNames = $this->getGroupChildNames($groupName, $callback);
         $layout = $this->getLayout();
-
         $childNamesSortOrder = [];
+        $sortedChildNames = [];
+        $notSortedChildNames = [];
 
         foreach ($groupChildNames as $childName) {
             $alias = $layout->getElementAlias($childName);
-            $sortOrder = (int)$this->getChildData($alias, 'sort_order') ?? 0;
 
-            $childNamesSortOrder[$sortOrder] = $childName;
+            if ($sortOrder = (int) $this->getChildData($alias, 'sort_order')) {
+                $childNamesSortOrder[$sortOrder][] = $childName;
```

Comment:
The issue when multiple items have the same sort order are replacing each other is still possible, isn't it? :)

Reply:
This issue is also fixed, by grouping the blocks by sort_order. So both issues should be fixed:
- blocks with the same sort_order
- blocks with no sort_order

Reply:
@dmytro-ch I believe if items have the same sort order, they will be stacked in `$childNamesSortOrder[$sortOrder][]`. So no override is happening here.

Reply:
@eduard13, @rogyar you're right!
Sorry for the misunderstanding.